corehq/apps/reports/filters/users.py

```diff
@@ -343,10 +349,14 @@ def pull_users_and_groups(cls, domain, mobile_user_and_group_slugs,
 
         user_dict = {}
         for group in groups:
-            user_dict["%s|%s" % (group.name, group._id)] = util.get_all_users_by_domain(
+            users_in_group = util.get_all_users_by_domain(
                 group=group,
                 simplified=True
             )
+            if limit_user_ids:
+                users_in_group = filter(lambda user: user['user_id'] in limit_user_ids, users_in_group)
+            user_dict["%s|%s" % (group.name, group._id)] = users_in_group
```

Comment:
The return value of this method is a little strange.  I just checked usages, and it looks like all but one use only `combined_users`, the remaining one uses `combined_users` and `admin_and_demo_users`.  It's probably out of scope to refactor this, but it's worth noting that the other properties don't appear to be used anywhere (I'd of course look closer before counting on that, though).